src/measure_theory/function/convergence_in_measure.lean

```diff
@@ -0,0 +1,364 @@
+begin
+  have hg : ae_measurable g μ, from ae_measurable_of_tendsto_metric_ae hf hfg,
+  refine tendsto_in_measure.congr (λ i, (hf i).ae_eq_mk.symm) hg.ae_eq_mk.symm _,
+  refine tendsto_in_measure_of_tendsto_ae_of_measurable (λ i, (hf i).measurable_mk)
+    hg.measurable_mk _,
+  have hf_eq_ae : ∀ᵐ x ∂μ, ∀ n, (hf n).mk (f n) x = f n x,
+    from ae_all_iff.mpr (λ n, (hf n).ae_eq_mk.symm),
+  filter_upwards [hf_eq_ae, hg.ae_eq_mk, hfg],
+  intros x hxf hxg hxfg,
+  rw [← hxg, funext (λ n, hxf n)],
+  exact hxfg,
+end
+
+namespace exists_seq_tendsto_ae
+
+lemma exists_nat_measure_lt_two_inv (hfg : tendsto_in_measure μ f at_top g) (n : ℕ) :
+  ∃ N, ∀ m ≥ N, μ {x | 2⁻¹ ^ n ≤ dist (f m x) (g x)} ≤ 2⁻¹ ^ n :=
+begin
+  specialize hfg (2⁻¹ ^ n) (by simp only [zero_lt_bit0, pow_pos, zero_lt_one, inv_pos]),
+  rw ennreal.tendsto_at_top_zero at hfg,
+  exact hfg (2⁻¹ ^ n) (pos_iff_ne_zero.mpr (λ h_zero, by simpa using pow_eq_zero h_zero))
+end
+
+/-- Given a sequence of functions `f` which converges in measure to `g`,
+`seq_tendsto_ae_seq_aux` is a sequence such that
+`∀ m ≥ seq_tendsto_ae_seq_aux n, μ {x | 2⁻¹ ^ n ≤ dist (f m x) (g x)} ≤ 2⁻¹ ^ n`. -/
+noncomputable
+def seq_tendsto_ae_seq_aux (hfg : tendsto_in_measure μ f at_top g) (n : ℕ) :=
+classical.some (exists_nat_measure_lt_two_inv hfg n)
+
+/-- Transformation of `seq_tendsto_ae_seq_aux` to makes sure it is strictly monotone. -/
+noncomputable
+def seq_tendsto_ae_seq (hfg : tendsto_in_measure μ f at_top g) : ℕ → ℕ
+| 0 := seq_tendsto_ae_seq_aux hfg 0
+| (n + 1) :=  max (seq_tendsto_ae_seq_aux hfg (n + 1))
+  (seq_tendsto_ae_seq n + 1)
+
+lemma seq_tendsto_ae_seq_succ (hfg : tendsto_in_measure μ f at_top g) {n : ℕ} :
+  seq_tendsto_ae_seq hfg (n + 1) =
+  max (seq_tendsto_ae_seq_aux hfg (n + 1)) (seq_tendsto_ae_seq hfg n + 1) :=
+by rw seq_tendsto_ae_seq
+
+lemma seq_tendsto_ae_seq_spec (hfg : tendsto_in_measure μ f at_top g)
+  (n k : ℕ) (hn : seq_tendsto_ae_seq hfg n ≤ k) :
+  μ {x | 2⁻¹ ^ n ≤ dist (f k x) (g x)} ≤ 2⁻¹ ^ n :=
+begin
+  cases n,
+  { exact classical.some_spec (exists_nat_measure_lt_two_inv hfg 0) k hn },
+  { exact classical.some_spec (exists_nat_measure_lt_two_inv hfg _) _
+      (le_trans (le_max_left _ _) hn) }
+end
+
+lemma seq_tendsto_ae_seq_strict_mono (hfg : tendsto_in_measure μ f at_top g) :
+  strict_mono (seq_tendsto_ae_seq hfg) :=
+begin
+  refine strict_mono_nat_of_lt_succ (λ n, _),
+  rw seq_tendsto_ae_seq_succ,
+  exact lt_of_lt_of_le (lt_add_one $ seq_tendsto_ae_seq hfg n) (le_max_right _ _),
+end
+
+end exists_seq_tendsto_ae
+
+/-- If `f` is a sequence of functions which converges in measure to `g`, then there exists a
+subsequence of `f` which converges a.e. to `g`. -/
+lemma tendsto_in_measure.exists_seq_tendsto_ae
+  (hfg : tendsto_in_measure μ f at_top g) :
+  ∃ ns : ℕ → ℕ, strict_mono ns ∧ ∀ᵐ x ∂μ, tendsto (λ i, f (ns i) x) at_top (𝓝 (g x)) :=
+begin
+  have h_lt_ε_real : ∀ (ε : ℝ) (hε : 0 < ε), ∃ k : ℕ, 2⁻¹ ^ (k - 1 : ℝ) < ε,
```

Comment:
Could you sketch briefly the proof in a comment at the beginning of the proof, to help the reader follow what is going on? In general, this is a good idea to do this for nontrivial proofs like this one.